### PR TITLE
EVA-1265 Store RefSNP information in accessioning service

### PR DESCRIPTION
**Pull request by jmmut**
this is only the writer for RSs. the writer that will be used in the import pipeline has to combine both this and the ss one.

### CODE REVIEW

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+public class DbsnpClusteredVariantWriterTest {
+    private static final int TAXONOMY_1 = 3880;
+
+    private static final int TAXONOMY_2 = 3882;
+
+    private static final long EXPECTED_ACCESSION = 10000000000L;
+
+    private static final long EXPECTED_ACCESSION_2 = 10000000001L;
+
+    private static final String CONTIG_1 = "contig_1";
```
Comment (andresfsilva) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java]: Delete all unused constants

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Test
+    public void failsOnDuplicateVariant() throws Exception {
+        ClusteredVariant clusteredVariant = new ClusteredVariant("assembly", TAXONOMY_1, "contig", START_1,
+                                                                 VARIANT_TYPE, VALIDATED);
+        DbsnpClusteredVariantEntity variant = new DbsnpClusteredVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(clusteredVariant), clusteredVariant);
+
+        thrown.expect(RuntimeException.class);
```
Comment (andresfsilva) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java]: I would prefer using the `expect` in the annotation as `@Test(expected = RuntimeException.class)`. With that we know easily which test cases are expected to throw an exception

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/ISubmittedVariant.java [hunk 3]
```diff
@@ -22,8 +22,11 @@
 import java.time.LocalDateTime;
 
 /**
- * Abstract representation of the fields that uniquely identify an accessioned submitted variant. Implemented by the
- * entity serialized into the database {@link SubmittedVariantEntity} and the message/DTO used by the REST API.
+ * Abstract representation of the fields that uniquely identify an accessioned submitted variant.
+ *
+ * Implemented by the basic bean {@link SubmittedVariant}, the entity for EVA accessions
+ * {@link SubmittedVariantEntity}, and the entities for DbSNP accessions (regular imported SSs and SSs without
```
[the remark this answers is not in the document]
Comment (jmmut) [reply eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/ISubmittedVariant.java]: yeah, maybe "SubSnps with and without coordinates" is even better. I don't know why I thought SSs was clear at all

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java (hunk 2 above)
Comment (jmmut) [reply eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java]: I won't open this discussion again :) there's more than one statement in the test, so I'll keep it as-is

File: eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/summary/DbsnpClusteredVariantSummaryFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    private DbsnpClusteredVariantSummaryFunction summaryFunction;
+
+    @Before
+    public void setUp() {
+        summaryFunction = new DbsnpClusteredVariantSummaryFunction();
+    }
+
+    @Test
+    public void summaryFunctionMustBeIdempotent() {
+        IClusteredVariant ClusteredVariant1 =
```
Comment (cyenyxe) [inline eva-accession-core/src/test/java/uk/ac/ebi/eva/accession/core/summary/DbsnpClusteredVariantSummaryFunctionTest.java]: Variable names shouldn't be capitalized. Applies to all the methods in this class.

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    private int taxonomyAccession;
+
+    private String contig;
+
+    private long start;
+
+    private VariantType type;
+
+    @Field("validated")
+    private Boolean validated;
```
Comment (cyenyxe) [inline eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java]: Is the annotation really necessary here?

File: eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/ClusteredVariant.java [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        if (!assemblyAccession.equals(that.assemblyAccession)) {
+            return false;
+        }
+        if (!contig.equals(that.contig)) {
+            return false;
+        }
+        if (type != that.type) {
+            return false;
+        }
+        return validated != null ? validated.equals(that.validated) : that.validated == null;
```
Comment (pabarcgar) [inline eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/ClusteredVariant.java]: I think 'validated' should not be one of the attributes identifying the variant, as is not being used in DbsnpClusteredVariantSummaryFunction. Unless we can find two different RSs in the same taxonomy, assembly, position, type... but different validation status, but then, we should think about adding it also to the summary function.

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java (hunk 5 above)
Comment (pabarcgar) [inline eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java]: Shouldn't we set the 'validated' default (let say false) to null to save space as we did with the SSs?

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        if (!assemblyAccession.equals(that.assemblyAccession)) {
+            return false;
+        }
+        if (!contig.equals(that.contig)) {
+            return false;
+        }
+        if (type != that.type) {
+            return false;
+        }
+        return validated != null ? validated.equals(that.validated) : that.validated == null;
```
Comment (pabarcgar) [inline eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java]: Again, validated is not in the summary function and maybe it should not be in the equals and hashCode.

File: eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        assertEquals(2, accessions.size());
+        assertEquals(EXPECTED_ACCESSION, (long) accessions.get(0).getAccession());
+        assertEquals(EXPECTED_ACCESSION_2, (long) accessions.get(1).getAccession());
+
+        assertEquals(firstClusteredVariant, accessions.get(0).getModel());
+        assertEquals(secondClusteredVariant, accessions.get(1).getModel());
+    }
+
+    @Test
+    public void failsOnDuplicateVariant() throws Exception {
```
Comment (cyenyxe) [inline eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriterTest.java]: Maybe also test that the flags don't have any effect in terms of considering a record duplicate?

File: eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java (hunk 7 above)
Comment (jmmut) [reply eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java]: > we agreed to use equals as just java-wise (for tests about all the fields and such). we have the summary function to work with the (bussiness logic of) identifying fields
